Source/Windows10/Prism.Windows/Navigation/INavigationParametersExtensions.cs

```diff
@@ -29,14 +29,14 @@ public static NavigationMode GetNavigationMode(this INavigationParameters parame
             return (parameters as INavigationParametersInternal).GetValue<NavigationMode>(nameof(NavigationMode));
         }
 
-        public static INavigationService GetNavigationService(this INavigationParameters parameters)
+        public static IPlatformNavigationService GetNavigationService(this INavigationParameters parameters)
```

Comment:
YOu should leave this as is. No need to return the full service. We expose platform methods via extension methods.

Reply:
Re-casting is the first thing we have to do every single time. This method is used in one place, only: ViewModel.OnNavigatedTo where we use methods like GoForward() and Refresh() which are only visible in the Platform version of this interface. The developer doesn't have a practical need for this interface.

Reply:
Developers can still call GoForward and Refresh off of the INavigationService interface